Content.Server/GameObjects/Components/Items/FloorTileItemComponent.cs

```diff
@@ -66,27 +53,20 @@ public void AfterAttack(AfterAttackEventArgs eventArgs)
             }
 
             var tileDef = (ContentTileDefinition)_tileDefinitionManager[tile.Tile.TypeId];
-            if (tileDef.IsSubFloor)
+            if (tileDef.IsSubFloor && attacked == null)
             {
                 var desiredTile = _tileDefinitionManager[_outputTile];
                 mapGrid.SetTile(eventArgs.ClickLocation, new Tile(desiredTile.TileId));
                 _entitySystemManager.GetEntitySystem<AudioSystem>().Play("/Audio/items/genhit.ogg", Owner);
-
-                CurrentStackCount--;
-                if (CurrentStackCount <= 0)
+                if (!Stack.Use(1))
```

Comment:
This won't actually do what you want it to. `StackComponent.Use` returns `false` if it *couldn't* remove from the stack anymore.

You'll need to manually check for count 0 after the use and delete then.

Reply:
Also, due to the placement of the if, a 0-size floortile will still place a tile down.
Granted we shouldn't be getting 0-size tiles but hey, it happened in SS13...

Reply:
Thankya